src/libsystemd-network/sd-dhcp-client.c

```diff
@@ -1367,16 +1367,15 @@ static int client_timeout_resend(
                 } else if (client->attempt >= client->max_attempts)
                         goto error;
 
+                client->request_sent = time_now;
                 break;
 
         case DHCP_STATE_SELECTING:
                 r = client_send_discover(client);
                 if (r < 0 && client->attempt >= client->max_attempts)
                         goto error;
 
-                if (client->rapid_commit)
```

Comment:
Why is this unnecessary? If client->rapid_commit is false, we didn't set the rapid_commit option in the DISCOVER query, right? Does this handle the case of invalid ACK response from the server?

Reply:
Actually the value is not used, at least currently, when rapid commit is not enabled. But, setting this does not hurt anything. Just for simplicity and safety.